gunicorn/arbiter.py

```diff
@@ -473,6 +473,8 @@ def manage_workers(self):
         Maintain the number of workers by spawning or killing
         as required.
         """
+        orig_num_workers = self.num_workers
+
         if len(self.WORKERS.keys()) < self.num_workers:
```

Comment:
So I think the right version of this patch that cannot ever say the same number twice is this:

```python
original_length = len(self.WORKERS)
# ...
if len(self.WORKERS) != original_length:
    # ...
```